core/src/test/scala/sttp/client3/testing/websocket/WebSocketStreamingTest.scala

```diff
@@ -90,17 +88,12 @@ trait WebSocketStreamingTest[F[_], S] extends ToFutureWrapper { outer: Suite wit
     val buffer = new AtomicReference[String]("")
     prepend(WebSocketFrame.text("1"))(functionToPipe {
       case WebSocketFrame.Text(payload, false, _) =>
-        buffer.accumulateAndGet(
-          payload,
-          new BinaryOperator[String] {
-            override def apply(t: String, u: String): String = t + u
-          }
-        )
+        val s = buffer.get()
+        buffer.set(s + payload)
```

Comment:
while it doesn't matter here, in general there's a difference between the two operations - `buffer.accumulateAndGet` is atomic, while get+set is not. Or was it causing problems?

Reply:
I know but it seems that `accumulateAndGet` is not convertible to JS:
`Referring to non-existent method scala.concurrent.impl.Promise$Link.accumulateAndGet`

Reply:
Ah ok :)